quad.go

```diff
@@ -0,0 +1,159 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package quad provides numeric integration of single variable functions by quadrature.
+//
+package quad
```

Comment:
I can be convinced otherwise but what about calling the package `quadrature`. It is perhaps because of my background but the abbreviation quad for me can represent many objects in numerical mathematics, primarily a quadrilateral. Also, what is the future for this package? I can imagine there could be quadrature formulas for integration of functions on quadrilaterals, triangles, etc. In optimize we went with the longer version instead of just `opt` and I think it was a good decision because `opt` can be too ambiguous. Just as `quad`. Opinions?